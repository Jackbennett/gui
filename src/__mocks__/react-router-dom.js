import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
