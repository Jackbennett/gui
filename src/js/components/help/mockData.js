export const helpProps = {
  demoArtifactLink: 'https://dgsbl4vditpls.cloudfront.net/mender-demo-artifact.mender',
  docsVersion: undefined,
  isHosted: false,
  isEnterprise: false,
  menderVersion: 'master',
  menderArtifactVersion: 'master'
};
