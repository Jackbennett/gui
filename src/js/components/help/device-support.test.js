import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import DeviceSupport from './device-support';
import { helpProps } from './mockData';
import { undefineds } from '../../../../tests/mockData';

describe('DeviceSupport Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <DeviceSupport {...helpProps} />
      </MemoryRouter>
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
