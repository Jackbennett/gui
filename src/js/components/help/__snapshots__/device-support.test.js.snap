// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceSupport Component renders correctly 1`] = `
<div>
  <h2>
    Device Support
  </h2>
  <p>
    The Mender client runs on the device in order to install software updates. Therefore, at minimum you need to install the client binary on the device and it must support the device OS and hardware architecture. This allows you to deploy application updates by using Update Modules.
    <br />
    For a detailed introduction to the different approaches to do this, please refer to our
     
    <a
      href="https://docs.mender.io/overview/device-support"
      rel="noopener noreferrer"
      target="_blank"
    >
      documentation on device support
    </a>
    .
  </p>
  <h2>
    Operating systems
  </h2>
  <p>
    Debian family OSes, such as Debian, Ubuntu and Raspberry Pi OS are officially supported. For devices running other types of Linux OSes such as Buildroot and OpenWRT, see existing integrations in the
     
    <a
      href="https://hub.mender.io/c/board-integrations/6"
      rel="noopener noreferrer"
      target="_blank"
    >
      Mender Hub community
    </a>
    . There you can also find information about integrations for the Yocto Project or
     
    <a
      href="https://docs.mender.io/system-updates-yocto-project"
      rel="noopener noreferrer"
      target="_blank"
    >
      follow our documentation
    </a>
     
    to integrate your own device.
  </p>
</div>
`;
